healthchecks_test.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"testing"
+	"github.com/stretchr/testify/assert"
+	queueConsumer "github.com/Financial-Times/message-queue-gonsumer/consumer"
+	"net/http"
+	"io/ioutil"
+	"strings"
+	"github.com/Financial-Times/message-queue-go-producer/producer"
+	"errors"
+)
+
+const (
+	kafkaTopicsResponseBody = "[\"Concept\", \"NativeCmsMetadataPublicationEvents\"]"
+	presentTopic = "Concept"
+	nonExistingTopic = "invalid"
+)
+
+type mockTransport struct {
+	responseStatusCode int
+	responseBody       string
+}
+
+type mockProducerInstance struct {
```

Comment:
I think that the unit tests are more readable and clean if you use the following libraries:
- https://golang.org/pkg/net/http/httptest/ (mock a http server)
- https://godoc.org/github.com/stretchr/testify/mock (mock a generic object)

You can find an example here: https://github.com/Financial-Times/native-ingester/blob/master/resources/healthcheck_test.go